src/cli/README.md

```diff
@@ -1169,6 +1169,27 @@ DNS response for example.com. - fd4c:9574:3720:2:0:0:5db8:d822 TTL:20456
 Done
 ```
 
+### dns resolve4 \<hostname\> \[DNS server IP\] \[DNS server port\] \[response timeout (ms)\] \[max tx attempts\] \[recursion desired (boolean)\]
+
+Send DNS query to obtain IPv4 address for given hostname.
+
+Requires `OPENTHREAD_CONFIG_DNS_CLIENT_NAT64_ENABLE`.
+
+The parameters after `hostname` are optional. Any unspecified (or zero) value for these optional parameters is replaced by the value from the current default config (`dns config`).
+
+This command requires a NAT64 prefix to be configured and present in Thread Network Data.
+
+The outputted addresses are the NAT64 IPv6 translated versions of the IPv4 addresses from the query response.
+
+```bash
+> route
+2001:db8:122:344::/64 sn med d800
+Done
+
+> dns resolve4 ipv6.google.com
+> DNS response for ipv6.google.com - 2001:db8:122:344:c0:2:2100:: TTL: 300
```

Comment:
Should we resolve `ipv4.google.com`?

Should we show a /96 prefix? And use an actual IPv4 address that `ipv4.google.com` resolves to today (i.e. 142.250.191.78)?

```suggestion
> dns resolve4 ipv4.google.com
> DNS response for ipv4.google.com - 2001:db8:122:344:c0:2:2100:: TTL: 300
```

Reply:
Good idea. Changed in new push.